secure/storage/src/tests/suite.rs

```diff
@@ -1,110 +1,180 @@
+        test(storage.as_mut());
+
+        // Test the on_disk implementation
+        let temp_path = TempPath::new();
+        let mut storage = Box::new(OnDiskStorage::new(temp_path.path().to_path_buf()));
+        test(storage.as_mut());
+    }
+}
 
-    let public = Policy::public();
-    let u64_value_0 = 5;
-    let u64_value_1 = 2322;
+/// A test for verifying VaultStorage properly implements the LibraSecureStorage API. This test
+/// depends on running Vault, which can be done by using the provided docker run script in
+/// `docker/vault/run.sh`
+#[test]
+#[ignore]
+fn execute_storage_tests_vault() {
+    let mut storage = Box::new(vault::setup_vault());
+    for test in STORAGE_TESTS.iter() {
+        vault::reset_vault_storage(storage.as_mut());
+        test(storage.as_mut());
+    }
+}
 
-    let mut rng = StdRng::from_seed([13u8; 32]);
-    let key_value_0 = Ed25519PrivateKey::generate_for_testing(&mut rng);
-    let key_value_1 = Ed25519PrivateKey::generate_for_testing(&mut rng);
+/// This test tries to get and set non-existent keys in storage and asserts that the correct
+/// errors are returned on these operations.
+fn test_get_set_non_existent(storage: &mut dyn Storage) {
+    let crypto_key = "Private Key";
+    let crypto_value = Value::Ed25519PrivateKey(create_ed25519_key_for_testing());
+    let u64_key = "U64 Key";
+    let u64_value = Value::U64(10);
 
     assert_eq!(
-        storage.get(KEY_KEY).unwrap_err(),
-        Error::KeyNotSet(KEY_KEY.to_string())
+        storage.get(crypto_key).unwrap_err(),
+        Error::KeyNotSet(crypto_key.to_string())
     );
     assert_eq!(
-        storage.get(U64_KEY).unwrap_err(),
-        Error::KeyNotSet(U64_KEY.to_string())
+        storage.get(u64_key).unwrap_err(),
+        Error::KeyNotSet(u64_key.to_string())
     );
-
     assert_eq!(
-        storage
-            .set(KEY_KEY, Value::Ed25519PrivateKey(key_value_0.clone()))
-            .unwrap_err(),
-        Error::KeyNotSet(KEY_KEY.to_string())
+        storage.set(crypto_key, crypto_value).unwrap_err(),
+        Error::KeyNotSet(crypto_key.to_string())
     );
     assert_eq!(
-        storage.set(U64_KEY, Value::U64(u64_value_0)).unwrap_err(),
-        Error::KeyNotSet(U64_KEY.to_string())
+        storage.set(u64_key, u64_value).unwrap_err(),
+        Error::KeyNotSet(u64_key.to_string())
     );
+}
+
+/// This test stores various key/value pairs in storage, updates them, retrieves the values and
+/// unwraps them to ensure the correct value types are returned.
+fn test_create_get_set_unwrap(storage: &mut dyn Storage) {
+    let crypto_key = "Private Key";
+    let crypto_private_1 = create_ed25519_key_for_testing();
+    let crypto_private_2 = create_ed25519_key_for_testing();
+    let crypto_value_1 = Value::Ed25519PrivateKey(crypto_private_1.clone());
+    let crypto_value_2 = Value::Ed25519PrivateKey(crypto_private_2.clone());
+
+    let u64_key = "U64 Key";
```

Comment:
if we use the same key everywhere, what value does it have not having this as a const?

Reply:
This was more a style choice than anything. Most of the data for each test is defined within the scope of each test (because of rust..), so I felt it cleaner to be consistent and leave all test data defined locally. But again, I don't feel that strongly :)